lib/std/fs.zig

```diff
@@ -795,22 +795,31 @@ pub const Dir = struct {
                         .kind = base.kind,
                     };
                 } else {
-                    self.stack.pop().iter.dir.close();
+                    var item = self.stack.pop();
+                    if (item.dirname_len != 0) {
```

Comment:
```suggestion
                    if (self.stack.len != 0) {
```
Both have the same effect but this conveys what property we're really interested in.